js/models/messages.js

```diff
@@ -1253,32 +1253,67 @@
       // if we did not sync or trigger a sync message for this specific message already
       const shouldTriggerSyncMessage =
         !isOurDevice &&
-        !isOpenGroupMessage &&
         !isMediumGroupMessage &&
         !this.get('synced') &&
         !this.get('sentSync');
 
       // A message is synced if we triggered a sync message (sentSync)
       // and the current message was sent to our device (so a sync message)
-      const shouldMarkMessageAsSynced =
-        isOurDevice && !isOpenGroupMessage && this.get('sentSync');
+      const shouldMarkMessageAsSynced = isOurDevice && this.get('sentSync');
 
-      // Handle the sync logic here
-      if (shouldTriggerSyncMessage) {
+      const isSessionOrClosedMessage = !isOpenGroupMessage;
+
+      if (!isOpenGroupMessage) {
         const contentDecoded = textsecure.protobuf.Content.decode(
           sentMessage.plainTextBuffer
         );
         const { dataMessage } = contentDecoded;
-        if (dataMessage) {
-          await this.sendSyncMessage(dataMessage);
-        }
-      } else if (shouldMarkMessageAsSynced) {
-        this.set({ synced: true });
-      }
-      if (!isOpenGroupMessage) {
         const primaryPubKey = await libsession.Protocols.MultiDeviceProtocol.getPrimaryDevice(
           sentMessage.device
         );
+
+        /**
+         * We should hit the notify endpoint for push notification only if:
+         *  • It's a one-to-one chat or a closed group
+         *  • The message has either text or attachments
+         */
+        const hasBodyOrAttachments = Boolean(
+          dataMessage &&
+            (dataMessage.body ||
+              (dataMessage.attachments && dataMessage.attachments.length))
+        );
+        const shouldNotifyPushServer =
+          hasBodyOrAttachments &&
+          isSessionOrClosedMessage &&
+          sentMessage.ttl ===
```

Comment:
We actually don't need this check now that we're checking for a body or attachments :)

Reply:
Oops. I matched the android way which was checking this field, https://github.com/loki-project/session-android-service/pull/84/files#diff-d6257378506ec22d884fbbb68f831aa2R29

I will fix it in a new PR